pyproject.toml

```diff
@@ -3,6 +3,6 @@ requires = [
     "setuptools",
     "setuptools_scm>=6.2",
     "jinja2>=2.10.3",
-    "numpy"
+    "numpy>=1.25",
```

Comment:
see https://github.com/liberfa/pyerfa/pull/122#discussion_r1517444449

Reply:
I think it's now an actual requirement since my patch is using a constant that's not defined in earlier versions.

Reply:
I do think that `NPY_ABI_VERSION` was always defined (if it's not I should change the pattern).

FWIW, I never quite understood what is the advantage/disavantage, but overall it shouldn't matter much?  For wheel builds you want it, but it's not technically a strict requirement for building.  But if someone builds with `--no-build-isolation` it is ignored anyway...

Reply:
Happy to remove it if that's what @mhvk prefers

Reply:
Let's keep it out - I'd like to introduce a minimum version only if it is actually needed.